client/src/components/related/ProductCard.jsx

```diff
@@ -1,65 +1,154 @@
 import React from 'react';
-// import '../css/RelatedProducts.css';
+import axios from 'axios';
 
 const stars = [1, 1, 1, .8, 0];
 
-const ProductCard = (props) => {
-  if (!props.product) {
-    return null;
+class ProductCard extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleActionButtonClick = this.handleActionButtonClick.bind(this);
+    this.handleProductClick = this.handleProductClick.bind(this);
+
+    this.state = {
+      product: undefined,
+      productImageURL: '',
+      alternateImageURLs: []
+    };
   }
-  const { product, selectedProduct, selectProduct, type, action, value } = props;
-  let hoverText, actionClass, actionSymbol;
 
-  const handleActionButtonClick = (event) => {
+  handleActionButtonClick(event) {
     event.stopPropagation();
-    action(product);
-  };
+    const { productId, action } = this.props;
+    action(productId);
```

Comment:
What is 'action'? From the context it looks like you are calling some native React method?

Reply:
Nah, it's from `props`. In Business Requirements it's called an "action button" so I figured the thing it does might as well be called `action`